external_project/models/project.py

```diff
@@ -0,0 +1,321 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Akretion (http://www.akretion.com)
+# Benoit Guillot <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, api, models
+from odoo.exceptions import Warning as UserError
+from odoo.tools.translate import _
+from odoo.tools.safe_eval import safe_eval
+from lxml import etree
+import requests
+import json
+
+
+ISSUE_DESCRIPTION = u"""Ce qui ne va pas:
+---------------------------
+
+
+Voilà comment cela devrait fonctionner:
+-----------------------------------------------------------
+"""
+
+
+class ExternalTask(models.Model):
+    _name = 'external.task'
+
+    @api.model
+    def _get_default_external_user(self):
+        return self.env.user
+
+    @api.model
+    def _get_origin(self):
+        if self.model_reference:
+            rec_name = self.model_reference._rec_name
+            if rec_name:
+                self.task_origin = self.model_reference.display_name
+
+    @api.model
+    def _authorised_models(self):
+        """ Inherit this method to add more models depending of your
+            modules dependencies
+        """
+        models = self.env['ir.model'].search(
+            [('model', '!=', 'external.task')])
+        return [(x.model, x.name) for x in models]
+
+    action_id = fields.Many2one(
+        'ir.actions.act_window', string="Action",
+        help="Action called to go to the original window.")
+    model_reference = fields.Reference(
+        selection='_authorised_models')
+    task_origin = fields.Char(compute='_get_origin')
+    name = fields.Char('Name')
+    stage_name = fields.Char('Stage')
+    description = fields.Text('Description', default=ISSUE_DESCRIPTION)
+    message_ids = fields.One2many(
+        comodel_name='external.message', inverse_name='res_id')
+    create_date = fields.Datetime('Create Date', readonly=True)
+    external_user_id = fields.Many2one(
+        comodel_name='res.users', string='Created by',
+        default=_get_default_external_user)
+    external_reviewer_id = fields.Many2one(
+        comodel_name='res.users', string='Reviewer')
+    assignee_name = fields.Char('Assignee name')
+
+    def get_url_key(self):
+        keychain = self.env['keychain.account']
+        if self.env.user.has_group('base.group_user'):
+            retrieve = keychain.suspend_security().retrieve
+        else:
+            retrieve = keychain.retrieve
+
+        account = retrieve(
+            [['namespace', '=', 'external_project']])[0]
+        return {
+            'url': account.login,
+            'api_key': account._get_password()
+        }
+
+    @api.model
+    def _call_odoo(self, reqtype, params, method=None, _id=None):
```

Comment:
Maybe to simplifier the code, we can do an endpoint that look like more to an odoo-rpc API. The idea will to always do json post request